modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotMetadataVerificationTask.java

```diff
@@ -88,25 +96,74 @@ public MetadataVerificationJob(SnapshotMetadataVerificationTaskArg arg) {
         }
 
         /** {@inheritDoc} */
-        @Override public List<SnapshotMetadata> execute() throws IgniteException {
+        @Override public List<SnapshotMetadata> execute() {
             IgniteSnapshotManager snpMgr = ignite.context().cache().context().snapshotMgr();
 
-            List<SnapshotMetadata> snpMeta = snpMgr.readSnapshotMetadatas(arg.snapshotName(), arg.snapshotPath());
+            List<SnapshotMetadata> metas = snpMgr.readSnapshotMetadatas(arg.snapshotName(), arg.snapshotPath());
+
+            for (SnapshotMetadata meta : metas)
+                checkMeta(meta, ignite.context().config().getEncryptionSpi().masterKeyDigest());
 
             if (arg.incrementIndex() > 0) {
-                List<SnapshotMetadata> metas = snpMeta.stream()
```

Comment:
Let's revert the name of the variable. You actually start using `metas` name for another variable. It's a little bit misleading.